propulsion-projector/Program.fs

```diff
@@ -230,6 +230,40 @@ module Args =
                 seconds x.Timeout, x.Retries, seconds x.MaxRetryWaitTime)
             let connector = Equinox.Cosmos.Connector(x.Timeout, x.Retries, x.MaxRetryWaitTime, Log.Logger, mode=x.Mode)
             discovery, x.Database, x.Container, connector
+#endif
+//#if sss
+    /// TODO: add DB connectors other than MsSql
+    type [<NoEquality; NoComparison>] SqlStreamStoreSourceParameters =
+        | [<AltCommandLine "-t"; Unique>]   Tail of intervalS: float
+        | [<AltCommandLine "-m"; Unique>]   BatchSize of int
+        | [<AltCommandLine "-c"; Unique>]   Connection of string
+        | [<AltCommandLine "-s">]           Schema of string
+        | [<AltCommandLine "-p"; Unique>]   Credentials of string
+        | [<Unique>]                        Checkpoints of string
+        interface IArgParserTemplate with
+            member a.Usage = a |> function
+                | Tail _ ->                 "Polling interval in Seconds. Default: 1"
+                | BatchSize _ ->            "Maximum events to request from feed. Default: 512"
+                | Connection _ ->           "Connection string for SqlStreamStore db. Optional if SQLSTREAMSTORE_CONNECTION specified"
+                | Schema _ ->               "Database schema name"
+                | Credentials _ ->          "Credentials string for SqlStreamStore db (Part of connection string, but NOT logged). Optional if SQLSTREAMSTORE_CREDENTIALS specified"
+                | Checkpoints _ ->          "Connection string for Checkpoints sql db. Optional if SQLSTREAMSTORE_CHECKPOINTS_CONNECTION specified. Default: same as `Connection`"
```

Comment:
Checkpoints is intended to be a potentially different connection string to a different db. So to be really prim and proper about it, it could also have the credentials part split out.

Reply:
You had me at prim...